### DISCUSSION TITLE
【读书会】深度学习 TensorFlow

### DISCUSSION THREAD

**Discussion by jiangzhonglian** [apachecn/home#229] [Ideas]
# <center>【读书会】深度学习 TensorFlow<center>

* 【推荐】深度学习 TensorFlow 从0～1入门
* TensorFlow 文档: <https://tensorflow.google.cn/tutorials>
* TensorFlow 视频: <https://developers.google.com/machine-learning/crash-course/ml-intro>

> 参与

* QQ群: 713436582
* 时间周期: 2018-12-10～2018-12-21 （共 14 天）
* 学习方式: 自主留言进度，无监督模式（主要是先过一遍，然后实现文档中的代码效果，有问题留言）

> 内容

1. [安装](https://tensorflow.google.cn/install/)
2. [学习（教程）](https://tensorflow.google.cn/tutorials/)
    1. [开始使用 TensorFlow](https://tensorflow.google.cn/tutorials/)
    2. 学习和使用机器学习
        * [概述](https://tensorflow.google.cn/tutorials/keras/)
        * [基本分类](https://tensorflow.google.cn/tutorials/keras/basic_classification)
        * [文本分类](https://tensorflow.google.cn/tutorials/keras/basic_text_classification)
        * [回归](https://tensorflow.google.cn/tutorials/keras/basic_regression)
        * [过拟合和欠拟合](https://tensorflow.google.cn/tutorials/keras/overfit_and_underfit)
        * [保存和恢复模型](https://tensorflow.google.cn/tutorials/keras/save_and_restore_models)

----

1. 学习快的，可以帮助协助回答问题
2. 学习快的，可以继续往下预习后面内容

----

安装: https://tensorflow.google.cn/install
学习: https://tensorflow.google.cn/tutorials
API：https://tensorflow.google.cn/guide  
 
![image](https://user-images.githubusercontent.com/9199175/50684826-e7de1f00-1051-11e9-88ae-b0052edd0929.png)

**Comment by jiangzhonglian** [comment]
## 学习计划 

> 用户: 片刻

| 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
| --- | --- | --- | --- | --- |
| 1  | 2018-12-10 | 安装                     | 100% |2018-12-10 |
| 2  | 2018-12-11 | 学习（教程）      | 100% | 2018-12-10 |
| 3  | 2018-12-12 | 概述                     | 100% | 2018-12-11 |
| 4  | 2018-12-14 | 基本分类              | 100% | 2018-12-11 |
| 5  | 2018-12-16 | 文本分类              | 100% | 2018-12-14 |
| 6  | 2018-12-18 | 回归                      | 100% | 2018-12-17 |
| 7  | 2018-12-20 | 过拟合和欠拟合   | 100% | 2018-12-17 |
| 8  | 2018-12-21 | 保存和恢复模型   | 100% | 2018-12-18 |

---

总结: [个人心得和体会]
1. 先看明白，有时间再测试运行！

Tensorflow 开发流程

* 1.加载包
* 2.导入数据集
* 3.探索数据
* 4.预处理数据
* 5.构建模型
  * 5.1设置层
  * 5.2编译模型
* 6.训练模型
* 7.评估准确率
* 8.做出预测


下面总结一下防止神经网络出现过拟合的最常见方法：

1. 获取更多训练数据。
2. 降低网络容量。
3. 添加权重正则化。
4. 添加丢弃层。

还有两个重要的方法在本指南中没有介绍：数据增强 和 批次归一化。

**Comment by jianfengMan** [comment]
## 学习计划 

> 用户: 小布

| 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
| --- | --- | --- | --- | --- |
| 1  | 2018-12-10 | 安装                     | 100% |2018-12-10 |
| 2  | 2018-12-11 | 学习（教程）      | 100% | 2018-12-10 |
| 3  | 2018-12-12 | 概述                     |100% | 2018-12-11 |
| 4  | 2018-12-14 | 基本分类              | 100% | 2018-12-11 |
| 5  | 2018-12-16 | 文本分类              |  100% | 2018-12-12|
| 6  | 2018-12-18 | 回归                      |  100% | 2018-12-13|
| 7  | 2018-12-20 | 过拟合和欠拟合   |  100% |2018-12-14 |
| 8  | 2018-12-21 | 保存和恢复模型   |  100% | 2018-12-15|
---
总结: [个人心得和体会]
跟着ApacheCN的脚步走，心理踏实。
Github: https://github.com/jianfengMan/mytensorflow

**Comment by icrn** [comment]
## 学习计划 

> 用户: icrn

| 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
| --- | --- | --- | --- | --- |
| 1  | 2018-12-10 | 安装                     | 100% |2018-12-10 |
| 2  | 2018-12-11 | 学习（教程）      | 100% | 2018-12-12|
| 3  | 2018-12-12 | 概述                      | 100% |2018-12-12 |
| 4  | 2018-12-14 | 基本分类              | 100% |2018-12-14 |
| 5  | 2018-12-16 | 文本分类              |  100% | 2018-12-15|
| 6  | 2018-12-18 | 回归                      |  0% | |
| 7  | 2018-12-20 | 过拟合和欠拟合   |  0% | |
| 8  | 2018-12-21 | 保存和恢复模型   |  0% | |

---

总结: [个人心得和体会]
代码跑了一遍。。。

**Comment by 1109226620** [comment]
## 学习计划 

> 用户: 我会发光

| 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
| --- | --- | --- | --- | --- |
| 1  | 2018-12-10 | 安装                     | 100% |2018-12-10 |
| 2  | 2018-12-11 | 学习（教程）      | 0% | |
| 3  | 2018-12-12 | 概述                      | 0% | |
| 4  | 2018-12-14 | 基本分类              | 0% | |
| 5  | 2018-12-16 | 文本分类              |  0% | |
| 6  | 2018-12-18 | 回归                      |  0% | |
| 7  | 2018-12-20 | 过拟合和欠拟合   |  0% | |
| 8  | 2018-12-21 | 保存和恢复模型   |  0% | |

---

总结: [个人心得和体会]
小白入门，pip直接安装，遇到一点问题百度解决了，还需要继续努力

**Comment by xiaoshuzh** [comment]
序号	完成时间	算法内容	完成进度	更新时间
1	2018-12-10	安装	100%	2018-12-11
2	2018-12-11	学习（教程）	100%	2018-12-11
3	2018-12-12	概述	100%	2018-12-11
4	2018-12-14	基本分类	100%	2018-12-11
5	2018-12-16	文本分类	0%	
6	2018-12-18	回归	0%	
7	2018-12-20	过拟合和欠拟合	0%	
8	2018-12-21	保存和恢复模型	0%	

之前已经学过了， 现在温习一波。谷歌官方的教程还是很清晰的，不错，希望可以坚持下去。

**Comment by lbllol365** [comment]
## 学习计划 



| 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
| --- | --- | --- | --- | --- |
| 1  | 2018-12-10 | 安装                     | 100% |2018-12-11 |
| 2  | 2018-12-11 | 学习（教程）      | 0% | |
| 3  | 2018-12-12 | 概述                      | 0% | |
| 4  | 2018-12-14 | 基本分类              | 0% | |
| 5  | 2018-12-16 | 文本分类              |  0% | |
| 6  | 2018-12-18 | 回归                      |  0% | |
| 7  | 2018-12-20 | 过拟合和欠拟合   |  0% | |
| 8  | 2018-12-21 | 保存和恢复模型   |  0% | |

---
之前学过一遍tf，但觉得学得不扎实，可能是用的少吧，所以跟着大家再学一遍

**Comment by wpfgogogo** [comment]
| 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
| --- | --- | --- | --- | --- |
| 1  | 2018-12-10 | 安装                     | 100% |2018-12-10 |
| 2  | 2018-12-11 | 学习（教程）      | 100% | 2018-12-10 |
| 3  | 2018-12-12 | 概述                    | 100% | 2018-12-11 |
| 4  | 2018-12-14 | 基本分类             | 100% | 2018-12-13 | 
| 5  | 2018-12-16 | 文本分类             | 100% | 2018-12-14 |
| 6  | 2018-12-18 | 回归                     | 100% | 2018-12-17 |
| 7  | 2018-12-20 | 过拟合和欠拟合  |  100% | | 2018-12-18 |
| 8  | 2018-12-21 | 保存和恢复模型   | 100% | | 2018-12-18 |

总结：
整体框架清楚了，下一步具体完善框架中的内容

**Comment by Lingmoumou** [comment]
## 学习计划 

  > 用户：brush
  
  | 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
  | --- | --- | --- | --- | --- |
  | 1  | 2018-12-10 | 安装                     | 100% |2018-12-10 |
  | 2  | 2018-12-11 | 学习（教程）      | 100% | 2018-12-12|
  | 3  | 2018-12-12 | 概述                      | 100% |2018-12-12 |
  | 4  | 2018-12-14 | 基本分类              | 0% | |
  | 5  | 2018-12-16 | 文本分类              |  0% | |
  | 6  | 2018-12-18 | 回归                      |  0% | |
  | 7  | 2018-12-20 | 过拟合和欠拟合   |  0% | |
  | 8  | 2018-12-21 | 保存和恢复模型   |  0% | |
  
  ---

 总结: Mark一下google的机器学习速成课程~

**Comment by qa8306202** [comment]
## 学习计划 

> 用户: ssozh

| 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
| --- | --- | --- | --- | --- |
| 1  | 2018-12-10 | 安装                     | 100% |2018-12-10 |
| 2  | 2018-12-11 | 学习（教程）      | 100% | 2018-12-12 |
| 3  | 2018-12-12 | 概述                     |100% | 2018-12-12 |
| 4  | 2018-12-14 | 基本分类              | 100% | 2018-12-12 |
| 5  | 2018-12-16 | 文本分类              | 100% |2018-12-14 |
| 6  | 2018-12-18 | 回归                      |  0% | |
| 7  | 2018-12-20 | 过拟合和欠拟合   |  0% | |
| 8  | 2018-12-21 | 保存和恢复模型   |  0% | |

---

总结: [个人心得和体会]
1. 今天是没有机会看明白了，不过总算是跑出来了，我用的方法是先下载下来zip文件，自己载入然后reshape55000*784=55000*28*28；不会的可以群里私@请叫我SsoZh

**Comment by HelWireless** [comment]
## 学习计划 

> 用户: 猫空

| 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
| --- | --- | --- | --- | --- |
| 1  | 2018-12-10 | 安装                     | 100% |2018-12-10 |
| 2  | 2018-12-11 | 学习（教程）      | 100% |2018-12-10 |
| 3  | 2018-12-12 | 概述                      | 100% |2018-12-11 |
| 4  | 2018-12-14 | 基本分类              | 100% | 2018-12-12|
| 5  | 2018-12-16 | 文本分类              |  100% |2018-12-14 |
| 6  | 2018-12-18 | 回归                      |  100% | 2018-12-18|
| 7  | 2018-12-20 | 过拟合和欠拟合   |  100% |2018-12-21 |
| 8  | 2018-12-21 | 保存和恢复模型   |  50% | 2018-12-21|

---

总结: [个人心得和体会]

对于checkpoint 还是有些不明白，基本的TensorFlow流程已经清晰，可以简单写分类回归的模型，后面找一两个例子试一试。

**Comment by anyuese** [comment]
用户:Warrior

序号	完成时间	算法内容	完成进度	更新时间
1	2018-12-10	安装	100%	2018-12-10
2	2018-12-11	学习（教程）	100%	2018-12-10
3	2018-12-12	概述	100%	2018-12-12
4	2018-12-14	基本分类	100%	2018-12-12
5	2018-12-16	文本分类	100%	
6	2018-12-18	回归	0%	
7	2018-12-20	过拟合和欠拟合	0%	
8	2018-12-21	保存和恢复模型
重新学了下numpy,pandas,matlpotib的使用

---

>> 友情提示: 
别回头去学，先明白这个教程关于TF的开发流程和思路，学完再去查漏补缺

**Comment by geek8565** [comment]
> | 序号 | 完成时间 | 算法内容 | 完成进度 | 更新时间 | 
> | --- | --- | --- | --- | --- |
> | 1  | 2018-12-10 | 安装                     | 100% |2018-12-12 |
> | 2  | 2018-12-11 | 学习（教程）      | 100% | 2018-12-12 |
> | 3  | 2018-12-12 | 概述                     |100% | 2018-12-13 |
> | 4  | 2018-12-14 | 基本分类              | 100% | 2018-12-13 |
> | 5  | 2018-12-16 | 文本分类              | 100% |2018-12-14 |
> | 6  | 2018-12-18 | 回归                      |  100% |2018-12-15 |
> | 7  | 2018-12-20 | 过拟合和欠拟合   |  100% |2018-12-15|
> | 8  | 2018-12-21 | 保存和恢复模型   | 100% |2018-12-15 |

讲实话，真心不错，封装的很完美，值得深入学习
